### ISSUE TITLE
Usage of CPU has changed from 4.5.2 to 4.6.0 notably

### ISSUE THREAD

**Issue by verdx** [wazuh/wazuh#18979]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.6.0-alpha1 | all | Manager and Agent | Packages | All |

When analysing the graphs for the [Syscollector Footprint test](https://github.com/wazuh/wazuh/issues/18928) for `4.6.0-alpha1`, and, comparing them with https://github.com/wazuh/wazuh/issues/18667, the CPU usage and the amplitude of it has changed in a significant way in all agents and the manager. 

<details><summary>Wazuh manager comparison</summary>

- 4.5.2
  ![image](https://github.com/wazuh/wazuh/assets/61122643/22f2a775-d347-45f2-a934-81f5bf9a1416)

- 4.6.0
  ![image](https://github.com/wazuh/wazuh/assets/61122643/44cb9b83-57a8-4e09-8eff-8feccbd2f960)

</details>

<details><summary>CentOS agent comparison</summary>

- 4.5.2
  ![image](https://github.com/wazuh/wazuh/assets/61122643/d65795bb-12d3-4d6c-865c-a0777b052539)

- 4.6.0
  ![image](https://github.com/wazuh/wazuh/assets/61122643/5427b3ca-4b9b-4bfa-8931-2b1423df270f)

</details>

<details><summary>Ubuntu agent comparison</summary>

- 4.5.2
  ![image](https://github.com/wazuh/wazuh/assets/61122643/8fd6b9ea-6f37-4bc5-8b9d-86c6da54507b)

- 4.6.0
  ![image](https://github.com/wazuh/wazuh/assets/61122643/292d2ced-df9b-44d8-b376-7352508cf220)

</details>

<details><summary>Windows agent comparison</summary>

- 4.5.2
  ![image](https://github.com/wazuh/wazuh/assets/61122643/1c6ea6e3-61e7-4177-9130-bbffafa431d9)

- 4.6.0
  ![image](https://github.com/wazuh/wazuh/assets/61122643/78f6e8eb-8073-4015-a4ef-07e3b5a21156)

</details>

**Comment by cborla** [comment]
## CPU utilization analysis

At first glance it can be seen that in version 4.6.0 the number of peaks is reduced, but the number of repetitions is increased, so we can assume that the CPU usage is distributed throughout the test instead of peaking at certain times. 

| Version | Total % CPU | Number of samples | Time between samples (seg) | Total time (seg) | % CPU-usage/seg |
|---|---|---|---|---|---|
| 4.5.2 | 68990.2 | 428728 | 5 | 2143640 | 0.0322425 |
| 4.6.0 | 69552.3 | 431432 | 5 | 2157160 | 0.0322184 |

After analyzing the issue, it seems that the difference in CPU usage compared to the previous version may not be relevant.
The table above gives an approximate amount of cpu % that the modulesd process used per second,  they are practically the same numbers, with an infimum difference of 0.0000241%.

![image](https://github.com/wazuh/wazuh/assets/41765669/c0fac30f-ee22-42a8-890a-bd7dee0d2948)

In conclusion, I interpret that there is no problem in cpu usage between the 2 versions.
